native/packages/react-native-bpk-component-carousel/src/BpkCarousel.js

```diff
@@ -0,0 +1,139 @@
+  ViewPropTypes,
+  StyleSheet,
+} from 'react-native';
+import BpkPaginationDots from 'react-native-bpk-component-pagination-dots';
+import { spacingXl } from 'bpk-tokens/tokens/base.react.native';
+import typeof BpkCarouselItem from './BpkCarouselItem';
+
+const styles = StyleSheet.create({
+  pagination: {
+    alignSelf: 'center',
+    marginTop: -spacingXl,
+  },
+});
+
+export type Props = {
+  accessibilityLabel: string | ((number, number) => string),
+  children: ChildrenArray<Element<BpkCarouselItem>>,
+  style: ?StyleObj,
+  renderPagination: ?boolean,
+};
+
+type State = {
+  currentIndex: number,
+  width: ?number,
+  height: ?number,
+};
+
+class BpkCarousel extends React.Component<Props, State> {
+  static propTypes = {
+    accessibilityLabel: PropTypes.oneOfType([PropTypes.string, PropTypes.func])
+      .isRequired,
+    children: PropTypes.node.isRequired,
+    style: ViewPropTypes.style,
+    renderPagination: PropTypes.bool,
+  };
+
+  static defaultProps = {
+    style: null,
+    renderPagination: true,
+  };
+
+  constructor(props: Props) {
+    super(props);
+    this.state = { width: null, height: null, currentIndex: 0 };
+  }
+
+  onLayout = (event: any) => {
+    this.setState({
+      width: event.nativeEvent.layout.width,
+      height: event.nativeEvent.layout.height,
+    });
+  };
+
+  getAccessibilityLabel() {
+    const { accessibilityLabel, children } = this.props;
+    if (typeof accessibilityLabel === 'function') {
+      return accessibilityLabel(
+        this.state.currentIndex,
+        React.Children.count(children),
+      );
+    }
+    return accessibilityLabel;
+  }
+
+  handleScroll = (event: any) => {
+    const { contentOffset } = event.nativeEvent;
+    const currentIndex = this.state.width
+      ? contentOffset.x / this.state.width
+      : 0;
+    this.setState({ currentIndex });
+  };
+
+  render() {
+    const { children, renderPagination, style } = this.props;
+    const { currentIndex, height, width } = this.state;
+
+    const shouldRenderChildren = width !== null && height !== null;
+    const childrenCount = React.Children.count(children);
+
+    return (
+      <Fragment>
+        <ScrollView
+          accessible
+          accessibilityLabel={this.getAccessibilityLabel()}
+          onMomentumScrollEnd={this.handleScroll}
+          onLayout={this.onLayout}
+          horizontal
+          pagingEnabled
+          showsHorizontalScrollIndicator={false}
+          style={style}
+        >
+          {shouldRenderChildren &&
+            React.Children.map(children, child =>
+              React.cloneElement(child, {
+                ...child.props,
+                style: [child.props.style, { width, height }],
+              }),
+            )}
+        </ScrollView>
+        {shouldRenderChildren &&
+          renderPagination && (
+            <BpkPaginationDots
+              style={styles.pagination}
+              accessibilityLabel={this.getAccessibilityLabel()}
```

Comment:
Not really sure this should be accessible (or what to put in here). The pagination is not focusable nor interactive, it doesn't seem something the screen reader should care about. What you guys think? @shaundon @matthewdavidson @k0nserv

Reply:
I tend to agree here - this is a purely visual indicator. The accessibility of a carousel should be handled by the scrollview itself. I can remove these props as part of my card.